src/rewrite_infinite_loops.cpp

```diff
@@ -0,0 +1,328 @@
+   (*source)->orig_prev_sp = 0;
+   set_chunk_parent(*source, parent_type);
+
+   *destination = *source;
+   *source      = next_source;
+}
+
+
+void rewrite_loop_keyword(Chunk *keyword, E_Token new_type)
+{
+   set_chunk_type(keyword, new_type);
+
+   switch (new_type)
+   {
+   case CT_DO:
+      keyword->orig_col_end += strlen("do") - keyword->Len();
+      keyword->str           = "do";
+      break;
+
+   case CT_WHILE:
+   case CT_WHILE_OF_DO:
+      keyword->orig_col_end += strlen("while") - keyword->Len();
+      keyword->str           = "while";
+      break;
+
+   case CT_FOR:
+      keyword->orig_col_end += strlen("for") - keyword->Len();
+      keyword->str           = "for";
+      break;
+
+   default:
+      assert(false);
+   }
+}
+
+
+static void rewrite_loop_condition(Chunk **source, Chunk **destination,
+                                   E_Token desired_type, const char *desired_condition)
+{
+   // Move the opening parenthesis
+   move_one_token(source, destination, desired_type);
+
+   // Move the condition
+   if (desired_type == CT_FOR)
+   {
+      set_chunk_type(*source, CT_SEMICOLON);
+      set_chunk_parent(*source, CT_FOR);
+      (*source)->str = ";";
+      move_one_token(source, destination, desired_type);
+      *destination = (*destination)->CopyAndAddAfter(*destination);
+   }
+   else
+   {
+      set_chunk_type(*source, CT_WORD);
+      (*source)->str = desired_condition;
+      move_one_token(source, destination, desired_type);
+   }
+
+   // If converting a 'for' to a 'while', delete the second semicolon
+   if (chunk_is_token(*source, CT_SEMICOLON))
+   {
+      Chunk *next_source = (*source)->GetNext();
+      Chunk::Delete(*source);
+      *source = next_source;
+   }
+   // Move the closing parenthesis
+   move_one_token(source, destination, desired_type);
+}
+
+
+void rewrite_loop_in_place(Chunk *keyword, E_Token desired_type, const char *desired_condition)
+{
+   Chunk *top    = keyword->GetNext();
+   Chunk *bottom = keyword;
+
+   rewrite_loop_keyword(keyword, desired_type);
+   rewrite_loop_condition(&top, &bottom, desired_type, desired_condition);
+}
+
+
+void rewrite_infinite_loops()
+{
+   LOG_FUNC_ENTRY();
+
+   E_Token    desired_type;
+   const char *desired_condition;
+
+   switch (options::mod_infinite_loop())
+   {
+   case 1: // for(;;)
+      desired_type      = CT_FOR;
+      desired_condition = nullptr;
+      break;
+
+   case 2: // while(true)
+      desired_type      = CT_WHILE;
+      desired_condition = "true";
+      break;
+
+   case 3: // do...while(true)
+      desired_type      = CT_WHILE_OF_DO;
+      desired_condition = "true";
+      break;
+
+   case 4: // while(1)
+      desired_type      = CT_WHILE;
+      desired_condition = "1";
+      break;
+
+   case 5: // do...while(1)
+      desired_type      = CT_WHILE_OF_DO;
+      desired_condition = "1";
+      break;
+
+   default:
+      assert(false);
```

Comment:
Don't 'assert', simply print a log message about the wrong value and return without doing any changes.